types/streaming.go

```diff
@@ -0,0 +1,29 @@
+package types
```

Comment:
I feel uncomfortable adding this into `types/`. That package is already too cluttered.

Reply:
What if it was in its own `stream/` sub-package in the `types/` directory? Similar to how there are the `grpc/` and `rest/` packages. My only hesitation to moving it into `baseapp/` is that there are very few interface definitions in that package, but it's not clear to me if that is by design or coincidence.